include the arch specific include dir for tags
use sort -u to remove duplicated file entries

diff --git a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -514,6 +514,7 @@ depend dep:	$(TIMESTAMP_FILE) $(VERSION_
 TAG_SUBDIRS = $(SUBDIRS)
 TAG_SUBDIRS += $(dir $(__LIBS))
 TAG_SUBDIRS += include
+TAG_SUBDIRS += arch/$(ARCH)/include
 
 FIND := find
 FINDFLAGS := -L
@@ -526,8 +527,8 @@ etags:
 		etags -a -o $(obj)etags `$(FIND) $(FINDFLAGS) $(TAG_SUBDIRS) \
 						-name '*.[chS]' -print`
 cscope:
-		$(FIND) $(FINDFLAGS) $(TAG_SUBDIRS) -name '*.[chS]' -print > \
-						cscope.files
+		$(FIND) $(FINDFLAGS) $(TAG_SUBDIRS) -name '*.[chS]' -print | \
+						sort -u > cscope.files
 		cscope -b -q -k
 
 SYSTEM_MAP = \
